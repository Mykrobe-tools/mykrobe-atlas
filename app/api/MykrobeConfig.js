import * as TargetConstants from 'constants/TargetConstants';

class MykrobeConfig {
  constructor() {
    this.targetName = require('../../package.json').targetName;

    if ('predictor-s-aureus' === this.targetName) {
      this.type = TargetConstants.TYPE_PREDICTOR;
      this.species = TargetConstants.SPECIES_S_AUREUS;
    }
    else if ('predictor-tb' === this.targetName) {
      this.type = TargetConstants.TYPE_PREDICTOR;
      this.species = TargetConstants.SPECIES_TB;
    }
    else {
      // default;
      this.type = TargetConstants.TYPE_PREDICTOR;
      this.species = TargetConstants.SPECIES_S_AUREUS;
    }
  }
}
export default MykrobeConfig;
