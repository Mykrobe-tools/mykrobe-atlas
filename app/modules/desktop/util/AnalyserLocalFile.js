/* @flow */

import EventEmitter from 'events';
import path from 'path';
import { spawn } from 'child_process';
import readline from 'readline';
import fs from 'fs-extra';

import { isString } from 'makeandship-js-common/src/utils/is';

import AnalyserJsonTransformer from '../../experiments/util/AnalyserJsonTransformer';
import * as APIConstants from '../../../constants/APIConstants';

import { pathToBin } from './pathToBin';
import { extensionForFileName } from '../../../util';
import isAnalyserError from './isAnalyserError';

const DEBUG =
  process.env.DEBUG_PRODUCTION === '1' ||
  process.env.NODE_ENV === 'development';

const tmp = require('tmp');
const app = require('electron').remote.app;

class AnalyserLocalFile extends EventEmitter {
  jsonBuffer: string;
  isBufferingJson: boolean;
  processExited: boolean;
  child: child_process$ChildProcess;
  didReceiveError: boolean;
  readLineExited: boolean;
  tmpObj: ?Object;

  analyseFile(filePaths: Array<string>, id: string = ''): AnalyserLocalFile {
    this.cancel();
    for (let i = 0; i < filePaths.length; i++) {
      const filePath = filePaths[i];
      const extension = extensionForFileName(filePath);
      if (extension === '.json') {
        return this.analyseJsonFile(filePath);
      }
      if (
        !APIConstants.API_SAMPLE_EXTENSIONS_ARRAY_WITH_DOTS.includes(extension)
      ) {
        const acceptable = APIConstants.API_SAMPLE_EXTENSIONS_ARRAY_WITH_DOTS.join(
          ', '
        );
        this.failWithError(
          `Mykrobe can only process files with extension: ${acceptable} - not ${extension}`
        );
        return this;
      }
    }
    return this.analyseBinaryFile(filePaths, id);
  }

  failWithError(err: string | Error) {
    if (!DEBUG) {
      // fail immediately
      this.cancel();
    }
    let message = err;
    if (err.message) {
      message = err.message;
    }
    setTimeout(() => {
      this.emit('error', {
        description: message,
      });
    }, 0);
  }

  doneWithJsonString(jsonString: string) {
    this.cleanup();
    const transformer = new AnalyserJsonTransformer();
    transformer
      .transform(jsonString)
      .then((result) => {
        // const { json, transformed } = result;
        // console.log('json', json);
        // console.log('transformed', transformed);
        this.emit('done', result);
      })
      .catch((err) => {
        this.failWithError(err);
      });
  }

  constructor() {
    super();
    app &&
      app.on('quit', () => {
        this.cancel();
        console.log('ev:app quit');
      });
  }

  analyseJsonFile = (file: File | string): AnalyserLocalFile => {
    if (isString(file)) {
      fs.readFile(file, (err, data) => {
        if (err) throw err;
        const dataString = data.toString();
        this.doneWithJsonString(dataString);
      });
    } else {
      const reader = new FileReader();
      reader.onload = () => {
        const dataString = reader.result;
        this.doneWithJsonString(dataString);
      };

      reader.onerror = (e) => {
        this.failWithError(
          `FileReader failed with error code ${e.target.error.code}`
        );
      };

      reader.readAsText(file);
    }
    return this;
  };

  analyseBinaryFile = (filePaths: Array<string>): AnalyserLocalFile => {
    this.tmpObj = tmp.dirSync({ prefix: 'mykrobe-' });

    const skeletonDir = path.join(this.tmpObj.name, 'skeleton');
    const tmpDir = path.join(this.tmpObj.name, 'tmp');

    fs.ensureDirSync(skeletonDir);
    fs.ensureDirSync(tmpDir);

    // Sample name should not contain whitespace, replace with '-'
    let sampleName = path.parse(filePaths[0]).name;
    sampleName = sampleName.replace(/\s+/g, '-');

    const pathToBinValue = pathToBin();

    // FIXME: if we don't use --force, occasionally get error - some files cached??

    const args = [
      'predict',
      '--force',
      sampleName,
      'tb',
      filePaths.length > 1 ? '--seq' : '-1',
      ...filePaths,
      '--tmp',
      tmpDir,
      '--skeleton_dir',
      skeletonDir,
      '--format',
      'json',
      '--guess_sequence_method',
      '--quiet',
    ];

    console.log('Guess at command line:', pathToBinValue, args.join(' '));
    console.log('Spawning executable at path:', pathToBinValue);
    console.log('With arguments:', args);
    const child = spawn(pathToBinValue, args);
    this.setChildProcess(child);

    if (DEBUG) {
      const tmpDir = tmp.dirSync({ prefix: 'mykrobe-debug-' }).name;
      const cmdPath = path.join(tmpDir, 'AnalyserLocalFile.cmd.txt');
      console.log('Logging command to', cmdPath);
      fs.writeFileSync(cmdPath, `${pathToBinValue} ${args.join(' ')}`);
    }

    return this.monitorChildProcess();
  };

  setChildProcess = (child: child_process$ChildProcess) => {
    this.child = child;
    return this;
  };

  monitorChildProcess = () => {
    this.jsonBuffer = '';
    this.isBufferingJson = false;
    this.processExited = false;
    this.readLineExited = false;

    if (!this.child) {
      this.failWithError('Failed to start child process');
      return this;
    }

    this.child.on('error', (err) => {
      console.log('Failed to start child process.', err);
      this.failWithError(err);
    });

    if (DEBUG) {
      const tmpDir = tmp.dirSync({ prefix: 'mykrobe-debug-' }).name;
      const stdoutPath = path.join(tmpDir, 'AnalyserLocalFile.stdout.txt');
      const stderrPath = path.join(tmpDir, 'AnalyserLocalFile.stderr.txt');
      console.log('Logging stdout and stderr to', stdoutPath, stderrPath);
      const stdoutStream = fs.createWriteStream(stdoutPath);
      const stderrStream = fs.createWriteStream(stderrPath);
      this.child.stdout.pipe(stdoutStream);
      this.child.stderr.pipe(stderrStream);
    }

    readline
      .createInterface({
        input: this.child.stdout,
      })
      .on('line', (line) => {
        if (this.didReceiveError) {
          return;
        }
        if (!this.isBufferingJson) {
          console.log(line);
        } else {
          console.log('Received json, muted');
        }
        if (line.indexOf('Progress:') >= 0) {
          console.log('progress');
          // we get a string like "[15 Oct 2017 16:19:47-Kac] Progress: 130,000/454,797"
          // extract groups of digits
          const trimmed = line.substr(line.indexOf('Progress:'));
          const digitGroups = trimmed.replace(/,/g, '').match(/\d+/g);
          if (digitGroups.length > 1) {
            const progress = parseInt(digitGroups[0]);
            const total = parseInt(digitGroups[1]);
            console.log('progress:' + progress);
            console.log('total:' + total);
            this.emit('progress', {
              progress,
              total,
            });
          }
        }

        if (this.isBufferingJson) {
          this.jsonBuffer += line;
        } else if (line.indexOf('{') !== -1) {
          // start collecting as soon as we see { in the buffer
          this.isBufferingJson = true;
          this.jsonBuffer = line.substr(line.indexOf('{'));
        }
      })
      .on('close', () => {
        console.log('readline stdin close');
        this.onReadLineExited();
      });

    readline
      .createInterface({
        input: this.child.stderr,
      })
      .on('line', (line) => {
        if (isAnalyserError(line)) {
          this.onReadLineError(line);
        } else {
          console.log('IGNORING ERROR: ' + line);
        }
      })
      .on('close', () => {
        console.log('readline stderr close');
      });

    this.child.on('exit', (code) => {
      console.log('Processing exited with code: ' + code);
      // 0 = exited normally, null = cancelled
      if (code === 0 || code === null) {
        this.onProcessExited();
      } else {
        this.failWithError(`Process exit unexpectedly with code ${code}`);
      }
    });

    return this;
  };

  onProcessExited = () => {
    this.processExited = true;
    this.checkDone();
  };

  onReadLineExited = () => {
    this.readLineExited = true;
    this.checkDone();
  };

  checkDone = () => {
    // wait until both readline and process have exited;
    // this can happen in different orders on different systems
    if (this.readLineExited && this.processExited && this.jsonBuffer.length) {
      this.doneWithJsonString(this.jsonBuffer);
    }
  };

  onReadLineError = (line: string) => {
    console.log('ERROR: ' + line);
    this.didReceiveError = true;
    this.failWithError(line);
  };

  cancel() {
    if (this.child) {
      this.child.kill();
      delete this.child;
    }
    this.cleanup();
  }

  cleanup() {
    console.log('cleanup');
    // this.tmpObj.removeCallback() doesn't always work
    this.tmpObj && fs.removeSync(this.tmpObj.name);
  }
}

export default AnalyserLocalFile;
