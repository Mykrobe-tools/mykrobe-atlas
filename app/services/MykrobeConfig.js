/* @flow */

import * as TargetConstants from '../constants/TargetConstants';

class MykrobeConfig {
  targetName: string;
  type: string;
  species: string;

  constructor() {
    this.targetName = require('../../package.json').targetName;

    // at the current time only TB is actively implemented
    // throw an error for any other version

    if (this.targetName === 'atlas-desktop-tb') {
      this.type = TargetConstants.TYPE_DESKTOP;
      this.species = TargetConstants.SPECIES_TB;
    } else if (this.targetName === 'atlas-tb') {
      this.type = TargetConstants.TYPE_WEB;
      this.species = TargetConstants.SPECIES_TB;
    } else {
      // target not recognised
      throw new Error(`Unsupported target name '${this.targetName}'`);
    }
  }

  isDesktop() {
    return this.type === TargetConstants.TYPE_DESKTOP;
  }

  isWeb() {
    return this.type === TargetConstants.TYPE_WEB;
  }
}
export default MykrobeConfig;
