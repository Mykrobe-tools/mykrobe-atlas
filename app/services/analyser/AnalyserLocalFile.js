/* @flow */

/* eslint-disable */

import path from 'path';
import fs from 'fs';
import os from 'os';
import { spawn } from 'child_process';
import * as TargetConstants from '../../constants/TargetConstants';
import AnalyserBaseFile from './AnalyserBaseFile';
import MykrobeConfig from '../MykrobeConfig';
const tmp = require('tmp');

// $FlowFixMe: Ignore Electron require
const app = require('electron').remote.app;
const platform = os.platform(); // eslint-disable-line global-require
const arch = os.arch();

class AnalyserLocalFile extends AnalyserBaseFile {
  jsonBuffer: string;
  isBufferingJson: boolean;
  processExited: boolean;
  child: child_process$ChildProcess; // eslint-disable-line camelcase
  didReceiveError: boolean;
  tmpObj: ?Object;

  constructor(targetConfig: MykrobeConfig) {
    super(targetConfig);
    app.on('quit', () => {
      this.cancel();
      console.log('ev:app quit');
    });
  }

  analyseBinaryFile(file: File): AnalyserLocalFile {
    // in Electron we get the full local file path
    // $FlowFixMe: Ignore missing type values
    const filePath = file.path;

    console.log('analyseBinaryFile', filePath);

    this.tmpObj = tmp.dirSync();
    const skeletonDir = path.join(this.tmpObj.name, 'skeleton');

    const fileName = path.parse(filePath).name;

    this.jsonBuffer = '';
    this.isBufferingJson = false;
    this.processExited = false;

    const pathToBin = this.pathToBin();
    const dirToBin = path.join(this.dirToBin(), this.targetConfig.targetName);
    const args = [
      'predict',
      '--force',
      `"${fileName}"`,
      'tb',
      '-1',
      `"${filePath}"`,
      '--skeleton_dir',
      `"${skeletonDir}"`,
    ];

    console.log('Spawning executable at path:', pathToBin);
    console.log('With arguments:', args);
    console.log('Guess at command line:', pathToBin, args.join(' '));
    this.child = spawn(pathToBin, args);

    if (!this.child) {
      this.failWithError('Failed to start child process');
      return this;
    }

    this.child.on('error', err => {
      console.log('Failed to start child process.', err);
      this.failWithError(err);
    });

    this.child.stdout.on('data', data => {
      if (this.didReceiveError) {
        return;
      }
      const dataString = data.toString('utf8');
      if ( !this.isBufferingJson ) {
        console.log(dataString);
      }
      else {
        console.log('Received json, muted');
      }
      if (dataString.indexOf('Progress:') >= 0) {
        console.log('progress');
        // we get a string like "[15 Oct 2017 16:19:47-Kac] Progress: 130,000/454,797"
        // extract groups of digits
        const trimmed = dataString.substr(dataString.indexOf('Progress:'));
        const digitGroups = trimmed.replace(/,/g,'').match(/\d+/g);
        if (digitGroups.length > 1) {
          const progress = parseInt(digitGroups[0]);
          const total = parseInt(digitGroups[1]);
          console.log('progress:'+progress);
          console.log('total:'+total);
          this.emit('progress', {
            progress,
            total
          });
        }
      }

      if (this.isBufferingJson) {
        this.jsonBuffer += dataString;
      } else if (dataString.indexOf('{') !== -1) {
        // start collecting as soon as we see { in the buffer
        this.isBufferingJson = true;
        this.jsonBuffer = dataString;
      }

      // sometimes receive json after process has exited
      if (this.isBufferingJson && this.processExited) {
        if (this.jsonBuffer.length) {
          console.log('done');
          this.doneWithJsonString(this.jsonBuffer);
        }
      }
    });

    this.child.stderr.on('data', data => {
      const dataString = data.toString('utf8');
      if ( dataString.startsWith('DEBUG') || dataString.startsWith('INFO') ) {
        console.log('IGNORING ERROR: ' + dataString);
        return;
      }
      this.didReceiveError = true;
      console.log('ERROR: ' + dataString);
      this.failWithError(dataString);
    });

    this.child.on('exit', code => {
      console.log('Processing exited with code: ' + code);
      // this.child = null;
      // deferring seems to allow the spawn to exit cleanly
      if (code === 0) {
        if (this.jsonBuffer.length) {
          console.log('done');
          this.doneWithJsonString(this.jsonBuffer);
        }
      }
      this.processExited = true;
    });

    return this;
  }

  cancel(): void {
    if (this.child) {
      this.child.kill();
      delete this.child;
    }
    this.tmpObj && this.tmpObj.removeCallback();
  }

  dirToBin() {
    const rootDir =
      process.env.NODE_ENV === 'development' ? process.cwd() : app.getAppPath();
    console.log('rootDir', rootDir);

    let dirToBin = '';

    if (process.env.NODE_ENV === 'development') {
      dirToBin = path.join(
        rootDir,
        `electron/resources/bin/${this.targetConfig.targetName}/${platform}-${arch}/bin`
      );
    } else {
      dirToBin = path.join(rootDir, '../bin');
    }
    console.log('dirToBin', dirToBin);
    return dirToBin;
  }

  pathToBin() {
    const UnsupportedError = new Error({
      message: 'Unsupported configuration',
      config: this.targetConfig
    });

    const dirToBin = this.dirToBin();

    let pathToBin = '';

    if (TargetConstants.TYPE_PREDICTOR === this.targetConfig.type) {
      if (TargetConstants.SPECIES_TB === this.targetConfig.species) {
        pathToBin = path.join(
          dirToBin,
          platform === 'win32' ? 'mykrobe_predictor.exe' : 'mykrobe_predictor'
        );
      }
      else {
        // unsupported configuration
        throw UnsupportedError;
      }
    } else {
      // unsupported configuration
      throw UnsupportedError;
    }

    console.log('pathToBin', pathToBin);

    return pathToBin;
  }
}

export default AnalyserLocalFile;
