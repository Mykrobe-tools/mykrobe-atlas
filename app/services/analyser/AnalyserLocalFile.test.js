import path from 'path';

import {
  ensurePredictorBinaries,
  ensureExemplarSamples,
  INCLUDE_SLOW_TESTS,
  EXEMPLAR_SAMPLES_FOLDER_PATH,
  expectCaseInsensitiveEqual,
} from '../../../desktop/util';

import AnalyserLocalFile from './AnalyserLocalFile';
import MykrobeConfig from '../MykrobeConfig';

const GENERATE_JSON_FIXTURES = true;

const exemplarSamplesExpect = require('../../../test/__fixtures__/exemplar-samples.expect.json');

// prerequisites

ensurePredictorBinaries();
ensureExemplarSamples();

const config = new MykrobeConfig();

INCLUDE_SLOW_TESTS && jest.setTimeout(10 * 60 * 1000); // 10 minutes

beforeEach(() => {
  process.env.NODE_ENV = 'development';
});

afterEach(() => {
  delete process.env.NODE_ENV;
});

describe('AnalyserLocalFile', () => {
  for (let i = 0; i < exemplarSamplesExpect.length; i++) {
    const exemplarSamplesExpectEntry = exemplarSamplesExpect[i];
    for (let j = 0; j < exemplarSamplesExpectEntry.source.length; j++) {
      const source = exemplarSamplesExpectEntry.source[j];
      const extension = source
        .substr(source.lastIndexOf('.') + 1)
        .toLowerCase();
      const isJson = extension === 'json';
      if (!isJson && !INCLUDE_SLOW_TESTS) {
        console.log(`Skipping slow test for ${source}`);
        continue;
      }
      it(`should analyse source file ${source}`, async done => {
        const analyser = new AnalyserLocalFile(config);
        const filePath = path.join(EXEMPLAR_SAMPLES_FOLDER_PATH, source);
        analyser
          .analyseFile(filePath)
          .on('progress', progress => {
            console.log('progress', progress);
          })
          .on('done', result => {
            const { json, transformed } = result;
            if (GENERATE_JSON_FIXTURES) {
              const fs = require('fs');
              // write unprocessed json
              fs.writeFileSync(
                `test/__fixtures__/exemplar-samples/${source}.json`,
                JSON.stringify(json, null, 2)
              );
              // write transformed json
              fs.writeFileSync(
                `test/__fixtures__/exemplar-samples/${source}__AnalyserLocalFile__.json`,
                JSON.stringify(transformed, null, 2)
              );
            }
            expectCaseInsensitiveEqual(
              transformed.speciesPretty,
              exemplarSamplesExpectEntry.expect.species
            );
            expectCaseInsensitiveEqual(
              transformed.resistant,
              exemplarSamplesExpectEntry.expect.all.resistant
            );
            expectCaseInsensitiveEqual(
              transformed.susceptible,
              exemplarSamplesExpectEntry.expect.all.susceptible
            );
            expectCaseInsensitiveEqual(
              transformed.drugsResistance,
              exemplarSamplesExpectEntry.expect.drugs.resistanceFlags
            );
            done();
          })
          .on('error', error => {
            // check if this was expected to be rejected
            let expectedReject = false;
            if (exemplarSamplesExpectEntry.expect.reject) {
              if (
                error.description &&
                error.description.includes(
                  'does not give susceptibility predictions'
                )
              ) {
                expectedReject = true;
              }
            }
            expect(expectedReject).toBeTruthy();
            if (!expectedReject) {
              throw error;
            }
            done();
          });
      });
    }
  }
});
