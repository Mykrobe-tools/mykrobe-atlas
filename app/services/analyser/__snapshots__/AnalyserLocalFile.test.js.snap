// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyserLocalFile should analyse and render a json file 1`] = `
<div
  className="container"
  data-tid="component-resistance-drugs"
>
  <div
    className="panelContainer"
    data-tid="panel-first-line-drugs"
    style={undefined}
  >
    <div
      className="panelContent"
    >
      <div
        className="panelTitle"
      >
        First line drugs
      </div>
      <div
        className="panelChildrenContainer"
      >
        <div
          className="drugs"
        >
          <div
            data-tid="drug"
          >
            Isoniazid
             
            <div
              className="susceptible"
            >
              Susceptible
            </div>
          </div>
          <div
            data-tid="drug"
          >
            Rifampicin
             
            <div
              className="resistant"
            >
              Resistant
            </div>
          </div>
          <div
            data-tid="drug"
          >
            Ethambutol
             
            <div
              className="resistant"
            >
              Resistant
            </div>
          </div>
          <div
            data-tid="drug"
          >
            Pyrazinamide
             
            <div
              className="susceptible"
            >
              Susceptible
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panelContainer"
    data-tid="panel-second-line-drugs"
    style={undefined}
  >
    <div
      className="panelContent"
    >
      <div
        className="panelTitle"
      >
        Second line drugs
      </div>
      <div
        className="panelChildrenContainer"
      >
        <div
          className="drugs"
        >
          <div
            data-tid="drug"
          >
            Quinolones
             
            <div
              className="resistant"
            >
              Resistant
            </div>
          </div>
          <div
            data-tid="drug"
          >
            Streptomycin
             
            <div
              className="susceptible"
            >
              Susceptible
            </div>
          </div>
          <div
            data-tid="drug"
          >
            Amikacin
             
            <div
              className="susceptible"
            >
              Susceptible
            </div>
          </div>
          <div
            data-tid="drug"
          >
            Capreomycin
             
            <div
              className="susceptible"
            >
              Susceptible
            </div>
          </div>
          <div
            data-tid="drug"
          >
            Kanamycin
             
            <div
              className="susceptible"
            >
              Susceptible
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AnalyserLocalFile should analyse and render a json file 2`] = `
<div
  className="container"
  data-tid="component-resistance-evidence"
>
  <div
    className="panelContainer"
    data-tid="panel-rifampicin"
    style={
      Object {
        "flex": "initial",
        "width": "50%",
      }
    }
  >
    <div
      className="panelContent"
    >
      <div
        className="panelTitle"
      >
        Rifampicin
      </div>
      <div
        className="panelChildrenContainer"
      >
        <div
          className="evidence"
        >
          <div
            data-tid="evidence"
          >
            Resistance mutation found: I491F in gene rpoB
          </div>
          <div
            data-tid="evidence"
          >
            Resistant allele coverage: 78
          </div>
          <div
            data-tid="evidence"
          >
            Susceptible allele coverage: 0
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panelContainer"
    data-tid="panel-ethambutol"
    style={
      Object {
        "flex": "initial",
        "width": "50%",
      }
    }
  >
    <div
      className="panelContent"
    >
      <div
        className="panelTitle"
      >
        Ethambutol
      </div>
      <div
        className="panelChildrenContainer"
      >
        <div
          className="evidence"
        >
          <div
            data-tid="evidence"
          >
            Resistance mutation found: M306I in gene embB
          </div>
          <div
            data-tid="evidence"
          >
            Resistant allele coverage: 77
          </div>
          <div
            data-tid="evidence"
          >
            Susceptible allele coverage: 0
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panelContainer"
    data-tid="panel-quinolones"
    style={
      Object {
        "flex": "initial",
        "width": "50%",
      }
    }
  >
    <div
      className="panelContent"
    >
      <div
        className="panelTitle"
      >
        Quinolones
      </div>
      <div
        className="panelChildrenContainer"
      >
        <div
          className="evidence"
        >
          <div
            data-tid="evidence"
          >
            Resistance mutation found: D94G in gene gyrA
          </div>
          <div
            data-tid="evidence"
          >
            Resistant allele coverage: 89
          </div>
          <div
            data-tid="evidence"
          >
            Susceptible allele coverage: 0
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AnalyserLocalFile should analyse and render a json file 3`] = `
<div
  className="container"
  data-tid="component-resistance-profile"
>
  <div
    className="column"
    data-tid="column-susceptible"
  >
    <div
      className="columnTitleSusceptible"
    >
      <i
        className="fa fa-check-circle"
      />
       
      Susceptible
    </div>
    <div
      data-tid="drug"
    >
      Capreomycin
    </div>
    <div
      data-tid="drug"
    >
      Isoniazid
    </div>
    <div
      data-tid="drug"
    >
      Amikacin
    </div>
    <div
      data-tid="drug"
    >
      Pyrazinamide
    </div>
    <div
      data-tid="drug"
    >
      Kanamycin
    </div>
    <div
      data-tid="drug"
    >
      Streptomycin
    </div>
  </div>
  <div
    className="column"
    data-tid="column-resistant"
  >
    <div
      className="columnTitleResistant"
    >
      <i
        className="fa fa-exclamation-triangle"
      />
       
      Resistant
    </div>
    <div
      data-tid="drug"
    >
      Rifampicin
    </div>
    <div
      data-tid="drug"
    >
      Ethambutol
    </div>
    <div
      data-tid="drug"
    >
      Quinolones
    </div>
  </div>
</div>
`;

exports[`AnalyserLocalFile should analyse and render a json file 4`] = `
<div
  className="container"
  data-tid="component-resistance-species"
>
  <div
    className="panelContainer"
    data-tid="panel-species"
    style={
      Object {
        "flex": "initial",
        "width": "100%",
      }
    }
  >
    <div
      className="panelContent"
    >
      <div
        className="panelTitle"
      >
        Species
      </div>
      <div
        className="panelChildrenContainer"
      >
        <div
          className="species"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<em>Mycobacterium tuberculosis</em> (lineage: European American)",
            }
          }
          tid="species"
        />
      </div>
    </div>
  </div>
</div>
`;
